toolchains/python/python.bzl

```diff
@@ -200,3 +200,184 @@ def nixpkgs_python_configure(
 
     if register:
         native.register_toolchains("@{}//:toolchain".format(name))
+
+
+def _nixpkgs_python_repository_impl(repository_ctx):
+    # 2. read generated json
+    python_modules = repository_ctx.read(repository_ctx.path(repository_ctx.attr.json_deps))
+
+    content = 'load("//:python_module.bzl", "python_module");'
+    for pkg_info in json.decode(python_modules):
+        pkg_name = pkg_info["name"]
+        pkg_store_path = pkg_info["store_path"]
+        deps = pkg_info["deps"]
+        pkg_link_path = "{}-link".format(pkg_name)
+        repository_ctx.symlink(pkg_store_path, pkg_link_path)
+
+        # Bazel chokes on files containing whitespaces, so we exclude them from
+        # the glob, hoping they are not important
+        content += """
+python_module(
+    name = "{name}",
+    store_path = "{link}",
+    files = glob(["{link}/**"], exclude=["{link}/**/* *"]),
+    deps = {deps},
+    visibility = ["//visibility:public"],
+)
+        """.format(name=pkg_name, link=pkg_link_path, deps=deps)
+
+    repository_ctx.file("BUILD.bazel", content)
+
+    # 3. generate BUILD.bazel file,
+    # ... _and_ the symlinks
+    # 4. Generate dummy WORKSPACE
+    # repository_ctx.file("WORKSPACE", "")
+
+    repository_ctx.file("python_module.bzl", """
+def _python_module_impl(ctx):
+    import_depsets = []
+    store = ctx.file.store_path
+    runfiles = ctx.runfiles(files = [store])
+
+    for dep in ctx.attr.deps:
+        runfiles = runfiles.merge(dep[DefaultInfo].data_runfiles)
+        import_depsets.append(dep[PyInfo].imports)
```

Comment:
[`data_runfiles` is deprecated and its use discouraged](https://bazel.build/extending/rules#deprecated_features), use `default_runfiles` and explicitly include required outputs instead.